lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -0,0 +1,286 @@
+        query_fragments = ['CREATE PUBLICATION %s FOR' % pg_quote_identifier(publication, 'publication')]
+        if tables:
+            table_quoted_list = []
+            for table in tables:
+                table_quoted_list.append('%s' % pg_quote_identifier(table, 'table'))
+                query_fragments.append('TABLE %s' % ', '.join(table_quoted_list))
+        else:
+            query_fragments.append('ALL TABLES')
+        query = ' '.join(query_fragments)
+        cursor.execute(query)
+        return True
+    else:
+        publication_info = get_publication_info(cursor, publication)
+        # If publication switch from targeted tables to all tables (or invert), recreate it
+        if publication_info['alltables'] is True and not tables or publication_info['alltables'] is True and not tables:
+            publication_delete(cursor, publication)
+            return publication_create(cursor, publication, tables)
+
+        return publication_modify_tables(cursor, publication, tables)
+
+
+def publication_matches(cursor, publication, owner):
+    if not publication_exists(cursor, publication):
+        return False
+    else:
+        publication_info = get_publication_info(cursor, publication)
+        if owner and owner != publication_info['owner']:
+            return False
+        else:
+            return True
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True),
+        database=dict(default="postgres"),
+        state=dict(default="present", choices=["absent", "present"]),
+        tables=dict(default=None, type="list")
+    ))
+    
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not postgresqldb_found:
+        module.fail_json(msg="the python psycopg2 module is required")
+
+    publication = module.params["name"]
+    state = module.params["state"]
+    database = module.params["database"]
+    tables = module.params["tables"]
+    changed = False
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port"
+    }
+    kw = dict((params_map[k], v) for (k, v) in module.params.items()
+              if k in params_map and v != '')
+
+    # If a login_unix_socket is specified, incorporate it here.
+    is_localhost = "host" not in kw or kw["host"] == "" or kw["host"] == "localhost"
+    if is_localhost and module.params["login_unix_socket"] != "":
+        kw["host"] = module.params["login_unix_socket"]
+
+    try:
+        db_connection = psycopg2.connect(database=database, **kw)
+        # Enable autocommit so we can create databases
```

Comment:
Is this needed in the context of that module ? I am not sure of the comment is out of date, or if that's the real reason and that do not apply to the module. If that's needed for another reason can it be changed ?

Reply:
discussing on #postgresql on IRC it seems i just need to update the commit, doing it